src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -421,104 +421,55 @@ public static void SetCommandTimeout(IDbCommand cmd)
             cmd.CommandTimeout = CachedServerInfo.Instance.GetQueryTimeoutSeconds(cmd.Connection);
         }
 
-
-        /// <summary>
-        /// Return true if the database is an Azure database
-        /// </summary>
-        /// <param name="connection"></param>
-        /// <returns></returns>
-        public static bool IsCloud(IDbConnection connection)
+        public static DatabaseEngineEdition GetEngineEdition(IDbConnection connection)
         {
             Validate.IsNotNull(nameof(connection), connection);
             if (!(connection.State == ConnectionState.Open))
             {
                 Logger.Write(TraceEventType.Warning, Resources.ConnectionPassedToIsCloudShouldBeOpen);
             }
 
-            Func<string, bool> executeCommand = commandText =>
+            Func<string, DatabaseEngineEdition> executeCommand = commandText =>
             {
-                bool result = false;
+                DatabaseEngineEdition result = DatabaseEngineEdition.Unknown;
                 ExecuteReader(connection,
-                          commandText,
-                          readResult: (reader) =>
-                          {
-                              reader.Read();
-                              int engineEditionId = int.Parse(reader[0].ToString(), CultureInfo.InvariantCulture);
-
-                              result = IsCloudEngineId(engineEditionId);
-                          }
+                    commandText,
+                    readResult: (reader) =>
+                    {
+                        reader.Read();
+                        result = (DatabaseEngineEdition)int.Parse(reader[0].ToString(), CultureInfo.InvariantCulture);
```

Comment:
we don't need to worry about this throwing?  or that's handled somewhere else?

Reply:
This script has existed in several places for a while now (i just changed the type we're casting to). We probably don't need to worry about it.